schema/proto3/common.proto

```diff
@@ -58,3 +60,10 @@ enum Protocol {
     HTTP = 3;
     ARP = 4;
 }
+
+enum NetworkType { 
+    VXLAN = 0; // the default type
+    VLAN = 1;
+    GRE = 2;
+    GENEVE = 3;
```

Comment:
Is it worth mentioning VXLAN-GPE here?

Reply:
I thought about that, the reasons I didn't put it there are:
1) Current neutron doesn't have support on VXLAN-GPE
2) It is not clear to me on which problem to solve by VLAN-GPE

Reply:
> 1. Current neutron doesn't have support on VXLAN-GPE

OVS should support VxLan-GPE already. The OVS commit for GPE support: https://openii.cn/openvswitch/ovs/commit/439f39cb9bd7214d1f70c317420cb2d33556296f

As Neutron doesn't support VxLan-GPE, actually this is a good opportunity. VxLan-GPE is more like a DP term but we could check what's required on CP to support it.

Reply:
I have added VxLAN-GPE option now after seeing the advantages: https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/cether/configuration/xe-16/ce-xe-16-book/vxlan-gpe-tunnel.html

Here is my analysis to support vxlan-gpe:
- VM will not need to know that tunnel type is used
- Agent will need to create the additional vxlan-gpe port during setup/init
- Agent will use the new vxlan-gpe port if NetworkType = VXLAN-GPE send down from Controller
- E2E functional testing and automated test code
- I don't see other changes needed unless we start leverage the vxlan-gpe specific features

We can say Alcor will have preliminary or proposed future support for VxLan-GPE. Thoughts?

Reply:
Sounds like a plan. One more question before we call it good: Can VxLan-GPE VNI overlap with "regular" VxLan VNI? Will this cause conflict on the host?

Reply:
My understanding is yes, the VxLan-GPE can use the same setup of VNI since it is backward compatible with VxLan. The official VxLan-GPE draft said VxLan-GPE vtep would detect if the other side is VxLan only then it would use VxLan encap header format. However, on the topic of how to detect vtep can support VxLan-GPE or it is VxLan only, the draft said it is out of scope...